src/zope/sqlalchemy/datamanager.py

```diff
@@ -37,6 +37,15 @@
     _retryable_errors.append(
         (psycopg2.extensions.TransactionRollbackError, None))
 
+try:
+    import psycopg.errors
+except ImportError:
+    pass
+else:
+    _retryable_errors.append(
+        (psycopg.errors.OperationalError, lambda e: e.sqlstate[:2] == '40')
+    )
```

Comment:
This code is unclear to me, could you please explain what case is handled here?

Reply:
This matches the `TransactionRollback` exception from psycopg2. psycopg only has specialized exceptions for each Class 40 error, but the shared superclass is `OperationalError`, so rather than adding five errors to the list, this seemed like a more economical check: https://www.psycopg.org/psycopg3/docs/api/errors.html (Class 40 - Transaction Rollback)